server/middleware/dbValidations.js

```diff
@@ -0,0 +1,71 @@
+import Pool from 'pg-pool';
+import validator from 'validator';
+import Helper from '../Helper/helper';
+import db from '../model/dbUsers';
+import dbloans from '../model/dbLoans';
+import dbrepayment from '../model/dbRepayment';
+import Message from '../Helper/responseMessage';
+
+const pool = new Pool({
+  user: 'victor_shagor',
+  host: 'localhost',
+  database: 'quickcredit',
+  password: 'oladimeji',
+  port: 5432,
+});
+
+const dbvalidate = {
+  /**
+  * @description verifies user input
+  * @param {object} req request parameter
+  * @param {object} res response object {status, data}
+  * @returns {object} error message
+  */
+  verifyInput(req, res, next) {
+    const requiredFields = ['firstName', 'lastName', 'email', 'address', 'password'];
+    const missingFields = [];
+    requiredFields.forEach((fields) => {
+      if (req.body[fields] === undefined) {
+        missingFields.push(fields);
+      }
+    });
+    if (missingFields.length !== 0) {
+      return res.status(400).send(
+        Message.errorMessage(400, `The following field(s) is/are required ${missingFields}`),
+      );
+    }
+    const {
+      firstName, lastName, email, address, password,
+    } = req.body;
+    if (!validator.isAlpha(firstName) || !validator.isAlpha(lastName)
+    || !validator.isLength(firstName, { min: 3 }) || !validator.isLength(lastName, { min: 3 })) {
+      return res.status(400).send(
+        Message.errorMessage(400, 'The names must be atleast 3 characters and can only be alphabets'),
+      );
+    }
+    if (!validator.isEmail(email)) {
+      return res.status(400).send(
+        Message.errorMessage(400, 'please enter a valid email address'),
+      );
+    }
+    if (!Helper.isValidEmail(password)) {
+      return res.status(400).send(
+        Message.errorMessage(400, 'Your password must contain atleast 8 characters and must atleast one number'),
+      );
+    }
+    if (!validator.isLength(address, { min: 5 })) {
+      return res.status(400).send(
+        Message.errorMessage(400, 'Your address must contain atleast 5 characters'),
+      );
+    }
+    pool.query('SELECT * FROM users WHERE email = $1 ', [email], (error, results) => {
```

Comment:
Consider requiring/selecting only the fields (columns) needed for this to work. Try as much as possible to avoid using the **"*"** in your queries for performance reasons. Always endeavour to be explicit and specify which result set you need to be returned from your queries.

Reply:
duly noted